### PR TITLE
Fix fr locales and enable tests in fr locale

### PR DESCRIPTION
**Pull request by fbonzon**
In long date formats `LL`, `LLL`, and `LLLL`, day of month 1 should be displayed with an ordinal:

    1er avril 1986

while days of month 2+ should be displayed with a cardinal:

    4 avril 1986

See #449

Tested:
yarn run lint
yarn run test

### CODE REVIEW

File: src/locale/fr/_lib/localize/index.js [hunk 1]
```diff
@@ -61,9 +61,9 @@ function ordinalNumber (dirtyNumber, dirtyOptions) {
 
   if (unit === 'isoWeek' || unit === 'week') {
     return feminineOrdinalNumber(number)
-  } else {
-    return masculineOrdinalNumber(number)
   }
+
```
Comment (fbonzon) [inline src/locale/fr/_lib/localize/index.js]: No functional change. This is for consistency with code in fr-CH locale.

File: src/locale/fr/test.js [hunk 2]
```diff
@@ -41,53 +41,53 @@ describe.skip('fr locale', function () {
...
-        assert(result === 'April')
+        assert(result === 'avril')
       })
     })
 
     describe('quarters', function () {
       it('Qo', function () {
-        var result = format(date, 'Qo [quarter]', {locale: locale})
-        assert(result === '2nd quarter')
+        var result = format(date, 'Qo [trimestre]', {locale: locale})
```
Comment (MBing) [inline src/locale/fr/test.js]: As 'trimestre' is more used as an academic concept, I would use 'quartier' here for more accounting purposes. As that would also be a better translation of what is used in the original version.

File: src/locale/fr/test.js [hunk 3]
```diff
@@ -41,53 +41,53 @@ describe.skip('fr locale', function () {
...
+        assert(result === '2e trimestre')
       })
     })
 
     describe('days of month', function () {
       it('Do', function () {
         var result = format(date, 'Do MMMM YYYY', {locale: locale})
-        assert(result === '4th April 1986')
+        assert(result === '4 avril 1986')
```
Comment (MBing) [inline src/locale/fr/test.js]: 4ème

File: src/locale/fr/test.js [hunk 4]
```diff
@@ -41,53 +41,53 @@ describe.skip('fr locale', function () {
...
+        assert(result === '1er jour de l’année')
       })
     })
 
     describe('days of week', function () {
       it('all variants', function () {
-        var result = format(date, 'do [day of the week,] dd ddd dddd', {locale: locale})
-        assert(result === '5th day of the week, Fr Fri Friday')
+        var result = format(date, 'do [jour de la semaine,] dd ddd dddd', {locale: locale})
+        assert(result === '5e jour de la semaine, ve ven. vendredi')
```
Comment (MBing) [inline src/locale/fr/test.js]: 5ème
I am not native French, so please let me know if I am wrong on this or what the exact rule is.
See also other numbers.

File: src/locale/fr/test.js (hunk 2 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Sorry, but "trimestre" in French is by far the most common translation of "quarter" in English.

Yes, the etymology is different. In French means 3 months and in English it's 1/4 of year, but that's how usage is. The literal translation is not good here.

File: src/locale/fr/test.js (hunk 3 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Can you motivate your suggestion?

Both ways to abbreviate ordinals are indeed seen. The shortest form is more frequent and recommended in all references I have. Just to cite one, the most important one (Académie Française):

http://www.academie-francaise.fr/abreviations-des-adjectifs-numeraux

> On écrit: La 8e femme de Barbe-bleue
> On n’écrit pas: La 8ème femme de Barbe-bleue

File: src/locale/fr/test.js (hunk 4 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Ditto. See previous comment.

File: src/locale/fr/test.js [hunk 5]
```diff
@@ -268,67 +302,67 @@ describe.skip('fr locale', function () {
...
     describe('months', function () {
       it('Mo', function () {
-        var result = parse('2014 12th', 'YYYY Mo', baseDate, {locale: locale})
+        var result = parse('2014 12e', 'YYYY Mo', baseDate, {locale: locale})
         assert.deepEqual(result, new Date(2014, 11 /* Dec */, 1))
       })
 
       it('MMM', function () {
-        var result = parse('2016 Nov', 'YYYY MMM', baseDate, {locale: locale})
+        var result = parse('2016 Nov.', 'YYYY MMM', baseDate, {locale: locale})
```
Comment (MBing) [inline src/locale/fr/test.js]: Months and days in French are all lowercase, even the abbreviations.

File: src/locale/fr/test.js [hunk 6]
```diff
@@ -268,67 +302,67 @@ describe.skip('fr locale', function () {
...
         assert.deepEqual(result, new Date(2016, 10 /* Nov */, 1))
       })
 
       it('MMMM', function () {
-        var result = parse('2016 December', 'YYYY MMMM', baseDate, {locale: locale})
+        var result = parse('2016 Décembre', 'YYYY MMMM', baseDate, {locale: locale})
```
Comment (MBing) [inline src/locale/fr/test.js]: Months and days in French are all lowercase.

File: src/locale/fr/test.js [hunk 7]
```diff
@@ -268,67 +302,67 @@ describe.skip('fr locale', function () {
...
 
       it('dd', function () {
-        var result = parse('2016 4 Mo', 'GGGG W dd', baseDate, {locale: locale})
+        var result = parse('2016 4 Lu', 'GGGG W dd', baseDate, {locale: locale})
         assert.deepEqual(result, new Date(2016, 0 /* Jan */, 25))
       })
 
       it('ddd', function () {
-        var result = parse('2016 4 Wed', 'GGGG W ddd', baseDate, {locale: locale})
+        var result = parse('2016 4 mer.', 'GGGG W ddd', baseDate, {locale: locale})
```
Comment (MBing) [inline src/locale/fr/test.js]: (Apology for repetitions, just to make it easier to find back.) ;)
Months and days in French are all lowercase, even the abbreviations.

File: src/locale/fr/test.js [hunk 8]
```diff
@@ -268,67 +302,67 @@ describe.skip('fr locale', function () {
...
     describe('days of a week', function () {
       it('do', function () {
-        var result = parse('2016 4 0th', 'GGGG W do', baseDate, {locale: locale})
+        var result = parse('2016 3 0e', 'GGGG W do', baseDate, {locale: locale})
         assert.deepEqual(result, new Date(2016, 0 /* Jan */, 24))
       })
 
       it('dd', function () {
-        var result = parse('2016 4 Mo', 'GGGG W dd', baseDate, {locale: locale})
+        var result = parse('2016 4 Lu', 'GGGG W dd', baseDate, {locale: locale})
```
Comment (MBing) [inline src/locale/fr/test.js]: (Apology for repetitions, just to make it easier to find back.) ;)
Months and days in French are all lowercase, even the abbreviations. Although this one seems of course something that isn't really official of course. Any thoughts on using lower/upper here?
Thanks!

File: src/locale/fr/test.js (hunk 5 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Done.

File: src/locale/fr/test.js (hunk 6 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Done.

File: src/locale/fr/test.js (hunk 8 above)
Comment (fbonzon) [reply src/locale/fr/test.js]: Done.

**Comment by fbonzon** [comment]
Changed to lowercase for consistency, even if – as mentioned by leshakoss – it doesn't matter for parsing.

**Pull request merged**